src/sympc/protocol/falcon/falcon.py

```diff
@@ -404,3 +405,52 @@ def multiplication_protocol(
             x.ring_size,
         )
         return z_value
+
+    @staticmethod
+    def select_shares(x: MPCTensor, y: MPCTensor, b: MPCTensor) -> MPCTensor:
+        """Returns either x or y based on bit b.
+
+        Args:
+            x (MPCTensor): input tensor
+            y (MPCTensor): input tensor
+            b (MPCTensor): input tensor which is shares of a bit used as selector bit.
+
+        Returns:
+            z (MPCTensor):Returns x (if b==0) or y (if b==1).
+
+        Raises:
+            ValueError: If the selector bit tensor is not of ring size "2".
+        """
+        ring_size = int(b.share_ptrs[0].get_ring_size().get_copy())
+        shape = b.shape
+        if ring_size != 2:
+            raise ValueError(
+                f"Invalid {ring_size} for selector bit,must be of ring size 2"
+            )
+        if shape is None:
```

Comment:
Q: Do we ever get here? Since every MPCTensor should have a shape

Reply:
I thought of a scenario, when we initialize MPCTensor, with shares (actual tensor values not pointers) ,I thought to sanity check for shape.In the long run , we would have to include a sanity checks in MPCTensor which automatically infers shape from the shares.

Reply:
Binary sharing we use it only internally only with pointers, I thought of scenario, where we might mix the ring_size and give input as shares values.